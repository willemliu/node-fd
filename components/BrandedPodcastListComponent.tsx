import styled from 'styled-components';
import { StyledH2, StyledAnchor, CardSection } from '../styles/Bnr';
import { ripple } from '../utils/ripple';
import Link from 'next/link';

export interface Props {
    title: string;
    items: any[];
}

export default (props: Props) => {
    return (
        <StyledList>
            <StyledH2>{props.title}</StyledH2>
            <ul>
                {props.items.map((item: any) => {
                    return (
                        <li key={item.id} className="ripple" onClick={ripple}>
                            <Link
                                href={`/article?articleId=${item.id}`}
                                as={item.publicationUrl}
                            >
                                <StyledAnchor
                                    href={`https://dev.bnr.nl${item.publicationUrl}`}
                                >
                                    {item.picture ? (
                                        <figure>
                                            <picture>
                                                <source
                                                    media="(max-width:640px)"
                                                    srcSet={
                                                        item.picture
                                                            .imageUrlSmall
                                                    }
                                                />
                                                <source
                                                    media="(max-width:860px)"
                                                    srcSet={
                                                        item.picture
                                                            .imageUrlMedium
                                                    }
                                                />
                                                <source
                                                    media="(min-width:861px)"
                                                    srcSet={
                                                        item.picture
                                                            .imageUrlLarge
                                                    }
                                                />
                                                <img
                                                    src={
                                                        item.picture
                                                            .baseImageUrl
                                                    }
                                                    alt={item.title}
                                                />
                                            </picture>
                                        </figure>
                                    ) : null}
                                    <CardSection>
                                        <h3>{item.title}</h3>
                                        <time>
                                            Duur: {item.durationInMinutes} min
                                        </time>
                                        <h3>{item.programTitle}</h3>
                                    </CardSection>
                                </StyledAnchor>
                            </Link>
                        </li>
                    );
                })}
            </ul>
        </StyledList>
    );
};

const StyledList = styled.section`
    ul {
        list-style: none;
        padding: 0;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        justify-content: space-between;
    }
    figure {
        margin: 0;
        img {
            width: 100%;
        }
    }
    li {
        flex: 0 1 calc(50% - 0.25rem);
        @media only screen and (min-width: 860px) {
            flex: 0 1 calc(25% - 0.25rem);
        }
        padding: 0;
        margin: 0 0 0.5rem 0;
        overflow: hidden;
        box-shadow: 0 2px 0.2px rgba(255, 0, 0, 1);
        border-radius: 5px;
        background-color: white;
        section {
            padding: 0.5rem;
        }

        animation: branded-glow 6s linear;
        animation-iteration-count: infinite;
    }

    @keyframes branded-glow {
        0% {
            box-shadow: 0 2px 0.2px 0px rgba(255, 0, 0, 1);
        }
        50% {
            box-shadow: 0 0 10px 0px rgba(255, 0, 0, 0.6);
        }
        85% {
            box-shadow: 0 2px 0.2px 0px rgba(255, 0, 0, 1);
        }
    }
`;
