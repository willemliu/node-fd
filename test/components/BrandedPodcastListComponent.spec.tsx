import React from 'react';
import { render } from '@testing-library/react';
import BrandedPodcastListComponent from '../../components/BrandedPodcastListComponent';

describe('Branded podcast list component', () => {
    it('should render branded podcasts correctly', () => {
        const { asFragment } = render(
            <BrandedPodcastListComponent
                title="Branded podcasts"
                items={[
                    {
                        id: 1,
                        title: 'title of the podcast',
                        publicationUrl: 'publication url',
                        programTitle: 'program title',
                        durationInMinutes: 3,
                        picture: {
                            imageUrlSmall: 'small image url',
                            imageUrlMedium: 'medium image url',
                            imageUrlLarge: 'large image url',
                            imageBaseUrl: 'base image url',
                        },
                    },
                    {
                        id: 2,
                        title: 'title of the podcast2',
                        publicationUrl: 'publication url2',
                        programTitle: 'program title2',
                        durationInMinutes: 4,
                        picture: {
                            imageUrlSmall: 'small image url2',
                            imageUrlMedium: 'medium image url2',
                            imageUrlLarge: 'large image url2',
                            imageBaseUrl: 'base image url2',
                        },
                    },
                ]}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
