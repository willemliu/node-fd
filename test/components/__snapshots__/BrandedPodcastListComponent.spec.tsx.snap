// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branded podcast list component should render branded podcasts correctly 1`] = `
<DocumentFragment>
  <section
    class="sc-ifAKCX btOyHJ"
  >
    <h2
      class="sc-bdVaJa EINmU"
    >
      Branded podcasts
    </h2>
    <ul>
      <li
        class="ripple"
      >
        <a
          class="sc-bwzfXH djPqxG"
          href="publication url"
        >
          <figure>
            <picture>
              <source
                media="(max-width:640px)"
                srcset="small image url"
              />
              <source
                media="(max-width:860px)"
                srcset="medium image url"
              />
              <source
                media="(min-width:861px)"
                srcset="large image url"
              />
              <img
                alt="title of the podcast"
                src="base image url"
              />
            </picture>
          </figure>
          <section
            class="sc-htpNat gpUwGh"
          >
            <h3>
              title of the podcast
            </h3>
            <time>
              Duur: 3 min
            </time>
            <h3>
              program title
            </h3>
          </section>
        </a>
      </li>
      <li
        class="ripple"
      >
        <a
          class="sc-bwzfXH djPqxG"
          href="publication url2"
        >
          <figure>
            <picture>
              <source
                media="(max-width:640px)"
                srcset="small image url2"
              />
              <source
                media="(max-width:860px)"
                srcset="medium image url2"
              />
              <source
                media="(min-width:861px)"
                srcset="large image url2"
              />
              <img
                alt="title of the podcast2"
                src="base image url2"
              />
            </picture>
          </figure>
          <section
            class="sc-htpNat gpUwGh"
          >
            <h3>
              title of the podcast2
            </h3>
            <time>
              Duur: 4 min
            </time>
            <h3>
              program title2
            </h3>
          </section>
        </a>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
